File: src/main/java/org/frc1778/freezylib/logging/SubscribedField.java
package org.frc1778.freezylib.logging;

import org.frc1778.freezylib.util.Measurement;

public class SubscribedField extends Field {

  private String value;

  public SubscribedField(String name, String unit, Class<?> type) {
    setName(name);
    setUnit(unit);
    this.value = String.valueOf(value);
    setType(type);
  }

  public SubscribedField(String name, Measurement.Base unit, Class<?> type) {
    this(name, unit.getSymbol(), type);
  }

  public <T> void pushValue(T value) {
    this.value = String.valueOf(value);
  }

  @Override
  public String getValue() {
    return value;
  }
}

File: src/main/kotlin/org/frc1778/freezylib/logging/SubscribedField.kt
package org.frc1778.freezylib.logging

import kotlin.reflect.KClass
import org.frc1778.freezylib.util.Measurement

class SubscribedField(name: String, unit: Measurement.Base, type: KClass<*>) : Field() {

    override var value = ""

    init {
        this.name = name
        this.unit = unit.symbol
        setType(type)
    }

    fun <T> pushValue(value: T) {
        this.value = value.toString()
    }
}
